### PR TITLE
Internship offer

### PR DESCRIPTION
**Pull request by dragoshan20**
get all offers from all companies task & create company internship offer task

### CODE REVIEW

File: src/main/java/ro/societateahermes/backendservice/entities/PurchasedPackage.java [hunk 1]
```diff
@@ -0,0 +1,5 @@
+package ro.societateahermes.backendservice.entities;
+public enum PurchasedPackage {
+
+    Silver,Gold,Bronze
```
Comment (saie2009) [inline src/main/java/ro/societateahermes/backendservice/entities/PurchasedPackage.java]: enums should be defined one under the other

File: src/main/java/ro/societateahermes/backendservice/service/serviceImplementation/OfferServiceImplementation.java [hunk 2]
```diff
@@ -0,0 +1,62 @@
...
+
+    @Override
+    public OfferDTO getOne(long offerId) {
+        return offerMapper.convertToDTO(offerRepository.findById(offerId).orElseThrow());
+    }
+
+    @Override
+    public List<OfferDTO> getBySponsor(long sponsorId) {
+        List<Offer> offerList = new ArrayList<>();
+        for (Offer o : offerRepository.findAll())
```
Comment (saie2009) [inline src/main/java/ro/societateahermes/backendservice/service/serviceImplementation/OfferServiceImplementation.java]: always use curly brackets

File: src/main/java/ro/societateahermes/backendservice/utils/mapper/OfferMapper.java [hunk 3]
```diff
@@ -0,0 +1,40 @@
...
+import org.springframework.stereotype.Component;
+import ro.societateahermes.backendservice.entities.Offer;
+import ro.societateahermes.backendservice.entities.dto.OfferDTO;
+import ro.societateahermes.backendservice.repository.EventRepositoryInterface;
+
+@Component
+public class OfferMapper {
+
+    private SponsorMapper sponsorMapper;
+    private EventRepositoryInterface eventRepo;
```
Comment (McShenanigans) [inline src/main/java/ro/societateahermes/backendservice/utils/mapper/OfferMapper.java]: Cine v-a dat ma ideea asta creata ca ar trebui sa aveti un repo in mapper? :)) Daca nu il folositi scoateti-l din clasa

Comment (Gabriela-Gabor) [reply src/main/java/ro/societateahermes/backendservice/utils/mapper/OfferMapper.java]: Aveam nevoie de el pentru sponsormapper:)))

Comment (McShenanigans) [reply src/main/java/ro/societateahermes/backendservice/utils/mapper/OfferMapper.java]: of bine :)) Am clarificat si cu gabi care-i treaba si e ok

**Review by McShenanigans — APPROVED** [review]

**Review by fujinuji — APPROVED** [review]

**Pull request merged**